src/asynk/async_queue.rs

```diff
@@ -139,43 +134,75 @@ pub trait AsyncQueueable {
 
     async fn fail_task(&mut self, task: Task, error_message: &str)
         -> Result<Task, AsyncQueueError>;
+
+    async fn fetch_periodic_tasks(
+        &mut self,
+        error_margin_seconds: i64,
+    ) -> Result<Option<Vec<PeriodicTask>>, AsyncQueueError>;
+
+    async fn insert_periodic_task(
+        &mut self,
+        metadata: serde_json::Value,
+        timestamp: DateTime<Utc>,
+        period: i32,
+    ) -> Result<PeriodicTask, AsyncQueueError>;
+    async fn schedule_next_task(
+        &mut self,
+        periodic_task: PeriodicTask,
+    ) -> Result<PeriodicTask, AsyncQueueError>;
 }
 
-#[derive(Debug, Clone)]
+#[derive(TypedBuilder, Debug, Clone)]
 pub struct AsyncQueue<Tls>
 where
     Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
     <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
     <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
     <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
 {
+    #[builder(setter(into))]
     pool: Pool<PostgresConnectionManager<Tls>>,
+    #[builder(setter(into))]
+    duplicated_tasks: bool,
```

Comment:
maybe a default value can be added for this field?

Reply:
default true ???

Reply:
False default

Reply:
Solve in [e1d31d2](https://github.com/ayrat555/fang/pull/47/commits/e1d31d215656077b385a07f4582a30a6d595a3c5)